pkg/web/apps/apps_psql_repository.go

```diff
@@ -19,16 +22,75 @@ func NewPostgreSQLRepository(db *sql.DB) Repository {
 
 // GetApps retrieves all apps from the database
 func (a *appsPostgreSQLRepository) GetApps() (*[]models.App, error) {
-	app1 := models.App{
-		ID:                    "a0874c82-2b7f-11e9-b210-d663bd873d93",
-		AppID:                 "com.aerogear.app1",
-		AppName:               "app1",
-		NumOfDeployedVersions: 1,
-		NumOfClients:          1,
-		NumOfAppLaunches:      1,
+	rows, err := a.db.Query(`
+	SELECT a.id,a.app_id,a.app_name,
+	COUNT(v.id) as num_of_deployed_versions, 
+	COALESCE(SUM(v.num_of_app_launches),0) as num_of_app_launches,
+	COALESCE(SUM(v.num_of_clients),0) as num_of_clients
+	FROM app as a 
+	LEFT JOIN version as v on a.app_id = v.app_id 
+	WHERE a.deleted_at IS NULL 
+	GROUP BY a.id;`)
+
+	if err != nil {
+		log.Error(err)
+		return nil, models.ErrInternalServerError
+	}
+
+	defer func() {
+		err := rows.Close()
+		if err != nil {
+			log.Error(err)
+		}
+	}()
+
+	apps := []models.App{}
+	for rows.Next() {
+		var a models.App
+		if err = rows.Scan(&a.ID, &a.AppID, &a.AppName, &a.NumOfDeployedVersions, &a.NumOfAppLaunches, &a.NumOfClients); err != nil {
+			log.Error(err)
+		}
+
+		apps = append(apps, a)
 	}
 
-	apps := []models.App{app1}
+	if len(apps) == 0 {
+		return nil, models.ErrNotFound
+	}
 
 	return &apps, nil
 }
+
+func (a *appsPostgreSQLRepository) GetAppVersionsByAppID(id string) (*[]models.Version, error) {
```

Comment:
I do not understand why you add this func here. You can create the struct Apps with just one query. In this same way, you can create just one query to return the struct app for some app_id. Also, I don't think it is part of this task anyway.

Reply:
@camilamacedo86 it is possible to retrieve all apps and versions but it does seem possible to map the rows from the `version` table into Go within this same query. It seems to be a limitation of the standard SQL Go library.

If you can find an example where this is done I would be happy to implement it, but have tried and could not find any.

Also, this is part of this task as this data is required for the `/apps` endpoint.

Reply:
Please, look my comment [here](https://github.com/aerogear/mobile-security-service/pull/42#discussion_r257296253)